File: backend/src/main/java/com/ptit/expensetracker/backend/ai/dto/context/TransactionDto.java
package com.ptit.expensetracker.backend.ai.dto.context;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TransactionDto {
    private Integer id;
    private Double amount;
    private String type; // INFLOW / OUTFLOW
    private String date; // ISO-8601 string (yyyy-MM-dd or datetime)
    private String description;
    private String categoryMetadata;
    private Integer walletId;
}

File: app/src/main/java/com/uet/expensetracker/features/ai/data/remote/dto/context/TransactionDto.kt
package com.uet.expensetracker.features.ai.data.remote.dto.context

data class TransactionDto(
    val id: Int?,
    val amount: Double?,
    val type: String?, // INFLOW / OUTFLOW
    val date: String?, // ISO-8601
    val description: String?,
    val categoryMetadata: String?,
    val walletId: Int?
)

